homeassistant/components/mpd/media_player.py

```diff
@@ -217,8 +217,13 @@ def media_content_type(self):
     @property
     def media_duration(self):
         """Return the duration of current playing media in seconds."""
-        # Time does not exist for streams
-        return self._currentsong.get("time")
+        time_from_status = self._status.get("time")
+        time = (
+            time_from_status.split(":")[1]
+            if isinstance(time_from_status, str) and ":" in time_from_status
+            else None
```

Comment:
We should avoid using multiline ternary statements, as it makes code hard to read.

Reply:
Fixed. @frenck can you re-review?